test/helpers/manifest.go

```diff
@@ -0,0 +1,273 @@
+func (n NamespaceName) String() string { return string(n) }
+
+// Manifest represents a deployment manifest that can consist of an any number
+// of Deployments, DaemonSets, Pods, etc.
+type Manifest struct {
+	// Filename is the file (not path) of the manifest. This must point to
+	// a file that contains any number of Deployments, DaemonSets, ...
+	Filename string
+
+	// NumDaemonSet is the number of DaemonSets in the manifest
+	NumDaemonSet int
+
+	// NumPods is the number of pods expected in the manifest, not counting
+	// any DaemonSets
+	NumPods int
+
+	// LabelSelector is the selector required to select *ALL* pods created
+	// from this manifest
+	LabelSelector string
+}
+
+// Deploy deploys the manifest. It will call ginkgoext.Fail() if any aspect of
+// that fails.
+func (m Manifest) Deploy(kubectl *Kubectl, namespace string) *Deployment {
+	deploy, err := m.deploy(kubectl, namespace)
+	if err != nil {
+		ginkgoext.Failf("Unable to deploy manifest %s: %s", m.Filename, err)
+	}
+
+	return deploy
+}
+
+func (m Manifest) deploy(kubectl *Kubectl, namespace string) (*Deployment, error) {
+	ginkgoext.By("Deploying %s in namespace %s", m.Filename, namespace)
+
+	numNodes := kubectl.GetNumCiliumNodes()
+	if numNodes == 0 {
+		return nil, fmt.Errorf("No available nodes to deploy")
+	}
+
+	path := ManifestGet(kubectl.BasePath(), m.Filename)
+	res := kubectl.Apply(ApplyOptions{Namespace: namespace, FilePath: path})
+	if !res.WasSuccessful() {
+		return nil, fmt.Errorf("Unable to deploy manifest %s: %s", path, res.CombineOutput().String())
+	}
+
+	d := &Deployment{
+		manifest:  m,
+		kubectl:   kubectl,
+		numNodes:  numNodes,
+		namespace: NamespaceName(namespace),
+		path:      path,
+	}
+
+	return d, nil
+}
+
+// Deployment is a deployed manifest. The deployment binds the manifest to a
+// particular namespace and records the number of nodes the deployment is
+// spread over.
+type Deployment struct {
+	kubectl   *Kubectl
+	manifest  Manifest
+	numNodes  int
+	namespace NamespaceName
+	path      string
+}
+
+// numExpectedPods returns the number of expected pods the deployment resulted
+// in
+func (d *Deployment) numExpectedPods() int {
+	return (d.numNodes * d.manifest.NumDaemonSet) + d.manifest.NumPods
+}
+
+// WaitUntilReady waits until all pods of the deployment are up and in ready
+// state
+func (d *Deployment) WaitUntilReady() {
+	expectedPods := d.numExpectedPods()
+	if expectedPods == 0 {
+		return
+	}
+
+	ginkgoext.By("Waiting for %s for %d pods of deployment %s to become ready",
+		HelperTimeout, expectedPods, d.manifest.Filename)
+
+	if err := d.kubectl.WaitforNPods(string(d.namespace), "", expectedPods, HelperTimeout); err != nil {
+		ginkgoext.Failf("Pods are not ready in time: %s", err)
+	}
+}
+
+// Delete deletes the deployment
+func (d *Deployment) Delete() {
+	ginkgoext.By("Deleting deployment %s", d.manifest.Filename)
+	d.kubectl.Delete(d.path)
```

Comment:
I think this delete will only delete objects in `default` namespace, we need namespaced version for `kubectl.Delete` here.

Reply:
Good catch.